### PR TITLE
DO NOT MERGE: Dockerfile and associated changes for NMN-DROP demo

### PR DESCRIPTION
**Pull request by nitishgupta**
PR contains dockerfile for running [nmn-drop](https://github.com/nitishgupta/nmn-drop) demo. 
Also added functionality to pass overrides when loading a model.

### CODE REVIEW

File: models.json [hunk 1]
```diff
@@ -16,6 +16,12 @@
         "predictor_name": "reading-comprehension",
         "max_request_length": 25819
     },
+    "nmn-drop": {
+        "archive_file": "https://nmn-drop.s3.amazonaws.com/model.tar.gz",
+        "predictor_name": "drop_demo_predictor",
+        "max_request_length": 25819,
+        "overrides": "{\"model\": { \"debug\":true}}"
```
Comment (schmmd) [inline models.json]: Eventually we'll want to have an `"image":` entry that corresponds to a built version of the Dockerfile in this PR.

**Comment by schmmd** [comment]
@nitishgupta how have you been using the included Dockerfile to date?

**Comment by schmmd** [comment]
@nitishgupta have you built and run the Dockerfile?  First I got:

```
Traceback (most recent call last):
  File "./app.py", line 36, in <module>
    from server.models import load_demo_models
  File "/stage/allennlp/server/models.py", line 7, in <module>
    import allennlp_rc
  File "/usr/local/lib/python3.6/site-packages/allennlp_rc/__init__.py", line 1, in <module>
    import allennlp_rc.dataset_readers
  File "/usr/local/lib/python3.6/site-packages/allennlp_rc/dataset_readers/__init__.py", line 1, in <module>
    from allennlp_rc.dataset_readers.drop import DropReader
  File "/usr/local/lib/python3.6/site-packages/allennlp_rc/dataset_readers/drop.py", line 25, in <module>
    from allennlp.data.tokenizers import Token, Tokenizer, SpacyTokenizer
ImportError: cannot import name 'SpacyTokenizer'
```

I noticed that the base image uses `allennlp/commit:ff0d44a5e21d5e6256c73b5b9f216a87c5743f91` but you install version `937d5945488a33c61d0047bd74d8106e60340bbd` of `allenai/allennlp-semparse` so I changed it to `FROM allennlp/commit:fa14bd8c177a12709a2fac0616eafaa9ad430b0a`.  However then I just got a different error:

```
$ docker run 90f8bb1b5b94
Traceback (most recent call last):
  File "./app.py", line 525, in <module>
    models = load_demo_models(args.models_file, args.model, model_names_only=args.no_models)
  File "/stage/allennlp/server/models.py", line 43, in load_demo_models
    util.import_submodules("semqa")
AttributeError: module 'allennlp.common.util' has no attribute 'import_submodules'
```

File: server/models.py [hunk 2]
```diff
@@ -38,6 +39,9 @@ def load_demo_models(models_file: str,
         model = blob[task_name]
         model_type = model.get("type", "allennlp")
 
+        if task_name == "nmn-drop":
+            util.import_submodules("semqa")
```
Comment (schmmd) [inline server/models.py]: When I build the Dockerfile I get an error at this line.  I don't see where `import_submodules` is defined.

```
Traceback (most recent call last):
  File "./app.py", line 525, in <module>
    models = load_demo_models(args.models_file, args.model, model_names_only=args.no_models)
  File "/stage/allennlp/server/models.py", line 43, in load_demo_models
    util.import_submodules("semqa")
AttributeError: module 'allennlp.common.util' has no attribute 'import_submodules'
```

**Comment by nitishgupta** [comment]
I didn't get the part "you install version `937d5945488a33c61d0047bd74d8106e60340bbd` of `allenai/allennlp-semparse` so I changed it to `FROM allennlp/commit:fa14bd8c177a12709a2fac0616eafaa9ad430b0a`

The `allennlp` version you changed it to is 6 days old and my code and the `allennlp-demo` Docker for my code is based off of `allennlp==0.9`. The `allennlp` commit I am using is from 24th Sept, 2019 and one before the version got bumped to `0.9`.

Without changing anything in my Docker, these are the commands I use and can run it:

```
export GIT_HASH=`git log -1 --pretty=format:"%H"`

docker build -f Dockerfile.nmn_drop -t nmn/demo:$GIT_HASH .

docker run -p 8000:8000 -v $HOME/.allennlp:/root/.allennlp --rm nmn/demo:$GIT_HASH --model nmn-drop
```

**Pull request closed without merging**